src/v/kafka/server/server.cc

```diff
@@ -347,6 +347,10 @@ ss::future<> server::apply(ss::lw_shared_ptr<net::connection> conn) {
 
     std::exception_ptr eptr;
     try {
+        co_await ctx->start();
```

Comment:
why not add something like

```
if (_hook && is_linked()) {
 ...
}
```

into ::stop? the same segfault would exist if start() were to throw before adding to the list.